src/components/atoms/NumberInput/BaseNumberInput.tsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+
+import { NumberInputProps } from './NumberInput';
+
+import { Container, Input } from './styles';
+
+
+const BaseNumberInput = React.forwardRef<HTMLInputElement, NumberInputProps>((props, ref) => {
+  const {
+    SuffixComponent = null,
+    PrefixComponent = null,
+    value = 0,
+    onChange,
+    min = Number.NEGATIVE_INFINITY,
+    max = Number.POSITIVE_INFINITY,
+    step = 1,
+    allowSpecialCharacters = false,
+    disableDecimal = false,
+    onKeyDown = () => {},
+    variant = 'default'
+  } = props;
+
+  const numberValue = Number(value);
+
+
+  const _onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    onChange(e);
+  };
+
+
+  const _onKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+
+    if ((allowSpecialCharacters && [ '+', '-', 'e' ].includes(e.key)) || (disableDecimal && [ '.' ].includes(e.key))) {
+      e.preventDefault();
+    }
+
+
+    //detect down arrow key
+    if (e.key === 'ArrowDown') {
```

Comment:
Can have a constant object containing all possible key codes which then can be used directly here.
And other people can use the object we export it from the library